```{r}

#libpath<-"/user_data/kalinka/r_packages"
#.libPaths(c(libpath, .libPaths()))
#remotes::install_github("MadsAlbertsen/ampvis2")
#install.packages("cowplot")
library(ggplot2)
library(ampvis2)
library(cowplot)
library(vroom)
library(tidyverse)

```

##Importing ampvis object
```{r}
amp<-readRDS("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_normalise_mmoX_pmoA.rds")
```

##### Subsetting to exclude "normalising_factor" ####

```{r}

tax_vector <- c("normalising_factor")
amp_f <- amp_subset_taxa(amp, tax_vector = tax_vector, remove = TRUE, normalise = TRUE)
amp_f$tax <- amp_f$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$"))
```


### Subsetting to remove the "unwanted" OTUs
```{r}



tax_vector_OTUs <- c("normalising_factor","Root; Homologues_mmoX","Root; likely_mmoX","Root_mmoX","Root_pmoA","amoA", "Nitrospira", "Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Root; pmoA_amoA_pxmA; pmoA", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus", "Root; pmoA_amoA_pxmA; pxmA", "Root; pmoA_amoA_pxmA")

f <- amp_subset_taxa(amp, tax_vector = tax_vector_OTUs, remove = TRUE, normalise = TRUE)
#x<-f$tax

f$tax <- f$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level

```


####Creating pca for all samples with and without the filter ####

```{r}

f_soil_sed <- amp_subset_samples(f,
  mfd_sampletype %in% c("Sediment", "Soil" ))

amp_soil_sed <- amp_subset_samples(amp_f,
  mfd_sampletype %in% c("Sediment", "Soil" ))

pca_all<- amp_ordinate(data = amp_soil_sed, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            opacity = 0.6,
            species_plot = TRUE,
           species_nlabels = 14,
           species_label_taxonomy = "Genus",
           species_label_size = 3,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - No filter")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca_f<- amp_ordinate(data = f_soil_sed, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            opacity = 0.6,
            species_plot = TRUE,
           species_nlabels = 14,
           species_label_taxonomy = "Genus",
           species_label_size = 3,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Filtered")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_all, pca_f, ncol = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_mmoX_pmoA.png",
       p,
        height = 5,
  width = 18)
```

##### Trying constraints #####

```{r}
f_urban <- amp_subset_samples(f,
  mfd_areatype %in% c("Urban"))


rda_fu<- amp_ordinate(data = f_urban, 
            type = "RDA", 
            transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            opacity = 0.6,
           constrain = c("mfd_hab1"),
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat Class. 2")+
scale_shape_discrete(name="Habitat Class. 1")+
  labs(title="Urban RDA - Constrained by Habitat Class. 1")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

rda_fu


 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/rda_urban_mmoX_pmoA.png",
       rda_fu,
        height = 6,
  width = 10)


pca_fu<- amp_ordinate(data = f_urban, 
            type = "PCA", 
            transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            opacity = 0.6,
                        species_plot = TRUE,
           species_nlabels = 17,
           species_label_taxonomy = "Genus",
           species_label_size = 3,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat Class. 2")+
scale_shape_discrete(name="Habitat Class. 1")+
  labs(title="Urban PCA")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_fu, rda_fu, nrow = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/urban_pca_rda_mmoX_pmoA.png",
       p,
        height = 8,
  width = 13)

pca_f<- amp_ordinate(data = f_soil_sed, 
            type = "PCA", 
            transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            opacity = 0.6,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Constrained by sampletype")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca_f

p<-plot_grid(pca_all, pca_f, ncol = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_mmoX_pmoA.png",
       p,
        height = 5,
  width = 18)
```



####### Diving into the natural soil samples ########

```{r}

m<-f$metadata

f_nat <- amp_subset_samples(f,
  mfd_areatype %in% c("Natural" ))


pca_f_nat<- amp_ordinate(data = f_nat, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            opacity = 0.6,
            species_plot = TRUE,
           species_nlabels = 14,
           species_label_taxonomy = "Genus",
           species_label_size = 3,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d","#666664","#0db2ee","#ee0d0d","#0d85ee","#eed70d","#0f0f0e","#2fee0d","#ee0dbd","#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Natural soil")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca_f_nat

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/nat_soil_pca.png",
       pca_f_nat,
        height = 10,
  width = 12)

f_meta_bog_mire_fen<-f$metadata%>%
filter(mfd_hab1 %in% c("Bogs, mires and fens"))%>%
filter(!is.na(mfd_hab2))

f_bog_mire_fen <- amp_load(otutable = f$abund, metadata = f_meta_bog_mire_fen, taxonomy = f$tax)

pca_f_bog_mire_fen<- amp_ordinate(data = f_bog_mire_fen, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            #sample_shape_by = "mfd_hab3",
            opacity = 0.6,
            species_plot = TRUE,
           species_nlabels = 30,
           species_label_taxonomy = "Genus",
           species_label_size = 3,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#697e47", "#913636", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Bogs, mires and fens")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))
pca_f_bog_mire_fen

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/bog_mire_fen_pca.png",
       pca_f_bog_mire_fen,
        height = 5,
  width = 10)

#### And a heatmap to match 

string<-f_bog_mire_fen$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])

hm_bog_mire <- amp_heatmap(f_bog_mire_fen,
  #facet_by = "mfd_hab1",
 group_by = "mfd_hab2",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5,
 round = 1,
 order_y_by = rev(genus_string),
 min_abundance=0.04,
 normalise = FALSE) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")
hm_bog_mire

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/bog_mire_fen_heatmap.png",
       hm_bog_mire,
        height = 7,
  width = 4.8)
```



######## Trying to do heatmap across all habitats in mfd level 2 ####
```{r}
f_meta_all_hab_2<-f$metadata%>%
filter(!is.na(mfd_hab2))
f_all_hab_2 <- amp_load(otutable = f$abund, metadata = f_meta_all_hab_2, taxonomy = f$tax)


f_all_hab_2$tax <- f_all_hab_2$tax %>%
  mutate(Genus = if_else(Class == "Methylococcaceae_mmoX","Methylococcaceae_mmoX", Genus))%>%
  mutate(Genus = if_else(Order == "Methylocystaceae","Methylocystaceae_mmoX", Genus))
tax<-f_all_hab_2$tax
string<-f_all_hab_2$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)

hm_all_hab_2 <- amp_heatmap(f_all_hab_2,
  facet_by = "mfd_areatype",
 group_by = "mfd_hab2",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("#f8f8f8", "darkred"),
 plot_colorscale = "log10",
 plot_values = FALSE,
 min_abundance=0.2,
order_y_by = rev(genus_string),
 normalise = FALSE) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")
hm_all_hab_2

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/hm_all_hab_2_red.png",
       hm_all_hab_2,
        height = 8,
  width = 15)
```

#### Testing Methylococcaceae_mmoX ######
```{r}


hm_all <- amp_heatmap(f_all_hab_2,
  #facet_by = "mfd_areatype",
 group_by ="sampling_comment",
 tax_aggregate = "Genus",
 tax_show = 10,
 color_vector = c("white", "darkred"),
 plot_colorscale = "log10",
 plot_values = FALSE,
 min_abundance=0.05,
 normalise = FALSE) +
theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
 axis.text.y = element_text(size=8),
  legend.position="right")
hm_all

ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/hm_all.png",
       hm_all,
        height = 5,
  width = 30)





#### Selecting the samples from long-read MFD ###

shallow_long_samples<-amp_filter_samples(f_all_hab_2,
                                    fieldsample_barcode %in% c("MFD01138", "MFD01223",
"MFD01248",
"MFD02979",
"MFD03346",
"MFD03399",
"MFD03638",
"MFD03726",
"MFD03899",
"MFD04408",
"MFD04434",
"MFD04967",
"MFD05580",
"MFD10064"), normalise = FALSE)
string<-long_samples$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)


hm_shallow_long_samples <- amp_heatmap(shallow_long_samples,
  #facet_by = "mfd_areatype",
 group_by ="fieldsample_barcode",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("#f8f8f8", "darkred"),
 plot_colorscale = "log10",
 plot_values = TRUE,
 plot_values_size = 2,
 round = 2,
 min_abundance=0.1,
order_y_by = rev(genus_string),
 normalise = FALSE) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")
hm_shallow_long_samples


ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/hm_long.png",
       hm_shallow_long_samples,
        height = 5,
  width = 7) 




#### And now to importing the combined count table from the long read samples ####

otutable_mmoX_long <-vroom("/user_data/kalinka/GraftM/long_read_graftM/mfd_deep_metagenomes/combined_count_table_mmoX.txt", delim = "\t")
otutable_mmoX_long <- otutable_mmoX_long %>%
rename(OTU = ConsensusLineage)  %>%
mutate(OTU = if_else(OTU == "Root", "Root_mmoX", OTU))
metadata <- readRDS("/projects/microflora_danica/Data_freezes/2022_10_10/2022-12-05_mfd-metadata.rds")
metadata <- metadata %>%
  rename(SeqId = fieldsample_barcode) %>%
    relocate(SeqId) %>%
  filter(after_total_reads > 4000000) %>% #Filtering at 4 million paired end reads
  drop_na(mfd_sampletype) ###Remove samples with no sample type
taxonomy <- data.frame(matrix(ncol = 8, nrow = 9))
x <- c("OTU","Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species")
colnames(taxonomy) <- x
taxonomy$OTU<-otutable_mmoX_long$OTU
taxonomy <- taxonomy %>%
separate(OTU, into = c("Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species"), sep = '; ', remove=FALSE)
taxonomy$Species<-taxonomy$OTU
amp_long<- amp_load(otutable = otutable_mmoX_long, metadata = metadata, taxonomy = taxonomy)

amp_long$tax <- amp_long$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) %>%
  mutate(Genus = if_else(Class == "Methylococcaceae_mmoX","Methylococcaceae_mmoX", Genus))%>%
  mutate(Genus = if_else(Order == "Methylocystaceae","Methylocystaceae_mmoX", Genus))
tax<-amp_long$tax
string<-amp_long$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)

### And then heatmapping it ####
long_heatmap <- amp_heatmap(amp_long,
  #facet_by = "mfd_areatype",
 group_by ="SeqId",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("#f8f8f8", "darkred"),
 plot_colorscale = "log10",
 plot_values = TRUE,
 plot_values_size = 2.5,
 round = 2,
 min_abundance=0.5,
order_y_by = rev(genus_string),
 normalise = FALSE) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8), legend.position="right")
long_heatmap

ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/hm_long_nanopore_mmoX.png",
       long_heatmap,
        height = 3,
  width = 7) 
#### Okay nej - det samme er slet ikke samlet op! Jeg skal altså finde ud af hvad der lige sker med de shallow prøver!
```


```



##################### Generating ampvis files ####################

```{r}
otutable_mmoX <-vroom("/user_data/kalinka/GraftM/shallow_21_12_2022/combined_count_table_mmoX.txt", delim = "\t")
otutable_mmoX <- otutable_mmoX %>%
rename(OTU = ConsensusLineage)  %>%
mutate(OTU = if_else(OTU == "Root", "Root_mmoX", OTU))

otutable_pmoA <-vroom("/user_data/kalinka/GraftM/shallow_21_12_2022/combined_count_table_pmoA.txt", delim = "\t")
otutable_pmoA <- otutable_pmoA %>%
rename(OTU = ConsensusLineage)%>%
mutate(OTU = if_else(OTU == "Root", "Root_pmoA", OTU))

combined <- bind_rows(otutable_pmoA, otutable_mmoX)
#Setting NA to 0
combined<- combined %>% replace(is.na(.), 0)


```

And adding a normalising factor in the form of number of reads
```{r}

normalising_factor<-metadata%>%
filter(after_total_bases > 1)%>%
mutate(normalising_factor=after_total_bases/(2*4500000))%>% #Genome size is 4.5 million
  select(SeqId, normalising_factor) %>%
  pivot_wider(names_from = SeqId, values_from = normalising_factor) %>%
  add_column(OTU="normalising_factor") %>%
  relocate(OTU)

combined_normalise <- bind_rows(combined, normalising_factor)
cols_to_remove <- names(combined_normalise)[colSums(is.na(combined_normalise)) > 0]

combined_normalise <- combined_normalise %>% select(-cols_to_remove)



```


Loading taxonomy data
```{r}

taxonomy <- data.frame(matrix(ncol = 8, nrow = 52))
x <- c("OTU","Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species")
colnames(taxonomy) <- x
taxonomy$OTU<-combined_normalise$OTU
taxonomy <- taxonomy %>%
separate(OTU, into = c("Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species"), sep = '; ', remove=FALSE)
taxonomy$Species<-taxonomy$OTU
```


And now loading the ampvis objects;
```{r}
amp_normalise<- amp_load(otutable = combined_normalise, metadata = metadata, taxonomy = taxonomy)

saveRDS(amp_normalise, file = "/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_normalise_mmoX_pmoA.rds")

```





















####### And then trying with the new metadata ######


```{r}
otutable <-vroom("/user_data/kalinka/GraftM/shallow_21_12_2022/combined_count_table_pmoA.txt", delim = "\t")
otutable <- otutable %>%
rename(OTU = ConsensusLineage)

metadata <- readRDS("/projects/microflora_danica/Data_freezes/2022_10_10/2022-12-05_mfd-metadata.rds")
metadata <- metadata %>%
  rename(SeqId = seq_id) %>%
    relocate(SeqId) %>%
  filter(after_total_reads > 4000000) %>% #Filtering at 4 million paired end reads
  drop_na(mfd_sampletype)


```

Importing taxonomy file and renameing the OTU-file
```{r}

taxonomy <- data.frame(matrix(ncol = 8, nrow = 37))
x <- c("OTU","Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species")
colnames(taxonomy) <- x
taxonomy$OTU<-otutable$OTU
taxonomy <- taxonomy %>%
separate(OTU, into = c("Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species"), sep = '; ', remove=FALSE)
taxonomy$Species<-taxonomy$OTU


#And a little trick to make species look like OTU-level;
```


```{r}
d <- amp_load(otutable = otutable, metadata = metadata, taxonomy = taxonomy)

saveRDS(d, file = "/user_data/kalinka/GraftM/shallow_21_12_2022/amp_pmoA_new_meta.rds")

```


And perhaps 1 big pca

```{r}

d_s <- amp_subset_samples(d,
  mfd_sampletype %in% c("Sediment", "Soil" ))


pca2<- amp_ordinate(data = d_s, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - New metadata")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca2

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_pmoA_new_graft_new_meta_24_01_23.png",
       pca2,
        height = 5,
  width = 10)

```

#### Okay and now to some filtering steps. I want to compare when I filter both the old and new data ######
```{r}
d_new<-readRDS("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_pmoA_new_meta.rds")
d_old <-readRDS("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_pmoA_old_graft_new_meta.rds")


d_new$tax <- d_new$tax %>%
  mutate(Species = if_else(Species == "Root", "Root_remove", Species))%>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level

d_old$tax <- d_old$tax %>%
  mutate(Species = if_else(Species == "Root", "Root_remove", Species))%>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level


tax_vector_new <- c("Root_remove","amoA", "Nitrospira", "Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Root; pmoA_amoA_pxmA; pmoA", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus", "Root; pmoA_amoA_pxmA; pxmA", "Root; pmoA_amoA_pxmA")

f_new <- amp_subset_taxa(d_new, tax_vector = tax_vector_new, remove = TRUE, normalise = TRUE)
x<-f_new$tax

tax_vector_old<-c("Root; pmoA_amoA_pxmA_umbrella; pxmA_amoA; pxmA","Rhodobacter_cluster","Root; pmoA_amoA_pxmA_umbrella","Root; pmoA_amoA_pxmA_umbrella; pmoA","Cycloclasticus_bradhyrhizobium_cluster","Root_remove","Total_Count", "HMO_cluster", "amoA", "Betaproteobacteria_amoA", "Actinobacteria", "Cycloclasticus", "Nitrosococcus", "Nitrospira_clade_A", "Nitrospira_clade_B")
f_old <- amp_subset_taxa(d_old, tax_vector = tax_vector_old, remove = TRUE, normalise = TRUE)
x<-f_old$tax

```

Og så til et lille plot
```{r}

f_new <- amp_subset_samples(f_new,
  mfd_sampletype %in% c("Sediment", "Soil" ))

f_old <- amp_subset_samples(f_old,
  mfd_sampletype %in% c("Sediment", "Soil" ))

pca_f_new<- amp_ordinate(data = f_new, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - New metadata and filtered")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca_f_old<- amp_ordinate(data = f_old, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_areatype",
            sample_shape_by = "mfd_sampletype",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Old GraftM package - New metadata and filtered")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_f_new, pca_f_old, ncol = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_old_and_new_with_filter.png",
       p,
        height = 6,
  width = 18)
```



What is happening to the agriculture???
```{r}

d_new<-readRDS("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_pmoA_new_meta.rds")
d_old <-readRDS("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/amp_pmoA_old_graft_new_meta.rds")

d_old$tax <- d_old$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level

d_new$tax <- d_new$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level



a_new <- amp_subset_samples(d_new,
  mfd_areatype %in% "Agriculture")

a_old <- amp_subset_samples(d_old,
  mfd_areatype %in% "Agriculture")


pca_a_new<- amp_ordinate(data = a_new, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Agriculture")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))


pca_a_new

pca_a_old<- amp_ordinate(data = a_old, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Old GraftM package - Agriculture")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_a_new, pca_a_old, ncol = 2)
p
 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_agriculture_no_filter.png",
       p,
        height = 6,
  width = 18)




af_new <- amp_subset_samples(f_new,
  mfd_areatype %in% "Agriculture")

af_old <- amp_subset_samples(f_old,
  mfd_areatype %in% "Agriculture")


pca_af_new<- amp_ordinate(data = af_new, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Agriculture after filter")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))


pca_af_new

pca_af_old<- amp_ordinate(data = af_old, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Old GraftM package - Agriculture after filter")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_af_new, pca_af_old, ncol = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_agriculture_after_filter.png",
       p,
        height = 6,
  width = 18)




##### Removed samples subset ######


d_new$tax <- d_new$tax %>%
  mutate(Species = if_else(Species == "Root", "Root_remove", Species))%>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level

d_old$tax <- d_old$tax %>%
  mutate(Species = if_else(Species == "Root", "Root_remove", Species))%>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level


tax_vector_new <- c("Root_remove","amoA", "Nitrospira", "Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Root; pmoA_amoA_pxmA; pmoA", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus", "Root; pmoA_amoA_pxmA; pxmA", "Root; pmoA_amoA_pxmA")
f_new_removed <- amp_subset_taxa(d_new, tax_vector = tax_vector_new, remove = FALSE, normalise = TRUE)

tax_vector_old<-c("Root; pmoA_amoA_pxmA_umbrella; pxmA_amoA; pxmA","Rhodobacter_cluster","Root; pmoA_amoA_pxmA_umbrella","Root; pmoA_amoA_pxmA_umbrella; pmoA","Cycloclasticus_bradhyrhizobium_cluster","Root_remove","Total_Count", "HMO_cluster", "amoA", "Betaproteobacteria_amoA", "Actinobacteria", "Cycloclasticus", "Nitrosococcus", "Nitrospira_clade_A", "Nitrospira_clade_B")
f_old_removed <- amp_subset_taxa(d_old, tax_vector = tax_vector_old, remove = FALSE, normalise = TRUE)

r_new <- amp_subset_samples(f_new_removed,
  mfd_areatype %in% "Agriculture")

r_old <- amp_subset_samples(f_old_removed,
  mfd_areatype %in% "Agriculture")

pca_r_new<- amp_ordinate(data = r_new, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Agriculture removed samples")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))


pca_r_old<- amp_ordinate(data = r_old, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "mfd_hab2",
            sample_shape_by = "mfd_hab1",
            species_plot = TRUE,
           species_nlabels = 10,
           species_label_taxonomy = "Genus",
           species_label_size = 4,
           species_label_color = "black",
           species_rescale = TRUE,
           filter_species=0,
            size=10) +
  scale_color_manual(values=c("#69d44f", "#ce1515", "#eed70d",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#01b5b7",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="Old GraftM package - Agriculture removed")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

p<-plot_grid(pca_r_new, pca_r_old, ncol = 2)

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_agriculture_removed.png",
       p,
        height = 6,
  width = 18)

### Heatmaps of the removed samples ####

## Using the all_subset just to see the difference in what is picked up between the two

a_new <- amp_subset_samples(f_new_removed,
  mfd_areatype %in% "Agriculture",  normalise = TRUE,  removeAbsentOTUs = TRUE )

r_old <- amp_subset_samples(f_old_removed,
  mfd_areatype %in% "Agriculture")


a_new <- amp_subset_samples(d_new,
  mfd_areatype %in% "Agriculture",  normalise = TRUE,  removeAbsentOTUs = TRUE)

a_old <- amp_subset_samples(d_old,
  mfd_areatype %in% "Agriculture",  normalise = TRUE,  removeAbsentOTUs = TRUE)


hm_r_new <- amp_heatmap(a_new,
 facet_by = "mfd_sampletype",
 group_by = "mfd_areatype",
 tax_aggregate = "Genus",
 tax_show = 18,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")

hm_r_new

hm_a_old <- amp_heatmap(af_old,
  facet_by = "mfd_hab1",
 group_by = "mfd_hab2",
 tax_aggregate = "Genus",
 tax_show = 18,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")

hm_a_old

```





################## With the old metadata #################

```{r}
otutable <-vroom("/user_data/kalinka/GraftM/shallow_21_12_2022/combined_count_table_pmoA.txt", delim = "\t")
otutable <- otutable %>%
rename(OTU = ConsensusLineage)

metadata <- read.delim("/projects/microflora_danica/Data_freezes/2022_09_08/2022-07-26_mfd_minimal_metadata.txt", header = TRUE, sep = " ") # nolint
metadata <- metadata %>%
  separate(gps_location, into = c("latitude","longitude"), sep = ',') %>%
  rename(SeqId = seq_id) %>%
    drop_na(after_total_reads) %>%
    relocate(SeqId) %>%
  filter(after_total_reads > 4000000)


```

Importing taxonomy file and renameing the OTU-file
```{r}
taxonomy <- read.delim("/user_data/kalinka/GraftM/combining_out/pmoA_taxonomy.csv", header = TRUE, sep = ",")
otutable$OTU<-taxonomy$OTU

taxonomy <- data.frame(matrix(ncol = 8, nrow = 37))
x <- c("OTU","Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species")
colnames(taxonomy) <- x
taxonomy$OTU<-otutable$OTU
taxonomy <- taxonomy %>%
separate(OTU, into = c("Kingdom", "Phylum", "Class","Order", "Family", "Genus", "Species"), sep = '; ', remove=FALSE)
taxonomy$Species<-taxonomy$OTU


#And a little trick to make species look like OTU-level;
```

```{r}
d <- amp_load(otutable = otutable, metadata = metadata, taxonomy = taxonomy)

saveRDS(d, file = "/user_data/kalinka/GraftM/shallow_21_12_2022/amp_pmoA_old_meta.rds")
#stats <- amp_alphadiv(data = d)

#amp_rarecurve(data = d, 
 #             color_by = "sample_type", 
  #            stepsize = 10) + 
   #           xlim(0, 900)

```


```{r}

pca_habitat<- amp_ordinate(data = d, 
             type = "PCA", 
             transform = "hellinger", 
             sample_color_by = "habitat_type")
            #sample_colorframe = TRUE)

pca_habitat

ggsave("/user_data/kalinka/metadata/pca_habitat.png",
       pca_habitat,
       height = 10,
       width = 10)

```


```{r}
heatmap_habitat<-amp_heatmap(d,
  group_by = "habitat_type",
  tax_aggregate = "Species",
  tax_show = 25,
  facet="habitat_type")+  
  theme(axis.text.x=element_blank(), axis.text.y = element_text(size = 9),
                    axis.ticks.x=element_blank())

ggsave("/user_data/kalinka/metadata/heatmap_habitat.png",
       heatmap_habitat,
       height = 8,
       width = 20)
```


New heatmap

```{r}

d_s <- amp_subset_samples(d,
  sample_type %in% c("sediment", "soil" ))

tax_vector<-c("Root_pmoA","Actinobacteria", "HMO_cluster")
d_no_root <- amp_subset_taxa(d_s, tax_vector = tax_vector, remove = TRUE)
  
heatmap2 <- amp_heatmap(d_s,
  facet_by = "sample_type",
 group_by = "habitat_type",
 tax_aggregate = "Species",
 tax_show = 32,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")

ggsave("/user_data/kalinka/metadata/heatmap2_pmoA.png",
       heatmap2,
       height = 9,
       width = 9)


heatmap2_no_root <- amp_heatmap(d_no_root,
  facet_by = "sample_type",
 group_by = "habitat_type",
 tax_aggregate = "Species",
 tax_show = 27,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")

ggsave("/user_data/kalinka/metadata/heatmap2_no_root_pmoA.png",
       heatmap2_no_root,
       height = 9,
       width = 9)

```


And some PCA stuff

```{r}

sediment_subset <- amp_subset_samples(d,
  sample_type %in% "sediment")

soil_subset <- amp_subset_samples(d,
   sample_type %in% "soil")

pca_soil<- amp_ordinate(data = soil_subset, 
       type = "PCA", 
           transform = "hellinger", 
           sample_color_by = "habitat_type")
pca_sediment<- amp_ordinate(data = sediment_subset, 
             type = "PCA", 
             transform = "hellinger", 

             sample_color_by = "habitat_type")


ggsave("/user_data/kalinka/metadata/pcasoil.png",
        pca_soil,
        height = 7,
        width = 8)
 ggsave("/user_data/kalinka/metadata/pcasediment.png",
       pca_sediment,
        pca_sediment,
        height = 7,
  width = 8)
```


And perhaps 1 big pca

```{r}
pca2<- amp_ordinate(data = d_s, 
            type = "PCA", 
             transform = "hellinger", 
            sample_color_by = "habitat_type",
            sample_shape_by = "sample_type",
            size=10) +
  scale_color_manual(values=c("#69d44f",
"#1f0060",
"#00c18d",
"#ec60c0",
"#004563",
"#8773ea",
"#d45828",
"#0092f3",
"#ce1515",
"#01b5b7",
"#eed70d",
"#dec7b0",
"#1d0b00"), name = "Habitat")+
scale_shape_discrete(name="Sample Type")+
  labs(title="GraftM package with contigs - Old metadata")+
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pca2

 ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/pca_pmoA_new_graft_old_meta_24_01_23.png",
       pca2,
        height = 5,
  width = 10)

```





Diversity
```{r}

tax_vector<-c("Total_Count", "HMO_cluster", "Root_pmoA", "amoA", "Betaproteobacteria_amoA", "Actinobacteria", "Cycloclasticus", "Nitrosococcus", "Nitrospira_clade_A", "Nitrospira_clade_B")

d_positives <- amp_subset_taxa(d, tax_vector = tax_vector, remove = TRUE, normalise = TRUE)

alphadiversityresult <- amp_alphadiv(d_positives,
  measure = c("observed","shannon"))


alphadiversityresult_subset<-alphadiversityresult%>%
filter(Shannon>2 & Reads>60) 



#Maybe I can make a new ampvis_element form this?

diversity_ampvis <- amp_load(otutable = otutable, metadata = alphadiversityresult_subset, taxonomy = taxonomy)
tax_vector<-c("Total_Count", "HMO_cluster", "Root_pmoA", "amoA", "Betaproteobacteria_amoA", "Actinobacteria", "Cycloclasticus", "Nitrosococcus", "Nitrospira_clade_A", "Nitrospira_clade_B")

diversity_ampvis <- amp_subset_taxa(diversity_ampvis, tax_vector = tax_vector, remove = TRUE, normalise = TRUE)

#That worked! Now to a little heatmap!



heatmap_diverse_samples <- amp_heatmap(diversity_ampvis,
  facet_by = "habitat_type",
 group_by = "fieldsample_barcode",
 tax_aggregate = "Species",
 tax_show = 27,
 color_vector = c("white", "darkred"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 2.5) +
theme(axis.text.x = element_text(angle = 45, size=8, vjust = 1),
 axis.text.y = element_text(size=8),
  legend.position="right")

heatmap_diverse_samples

ggsave("/user_data/kalinka/metadata/diverse.png",
       heatmap_diverse_samples,
        height = 8,
  width = 30)


```



#### Store Økssø


```{r}

tax_vector_OTUs <- c("normalising_factor","Root; Homologues_mmoX","Root; likely_mmoX","Root_mmoX","Root_pmoA","amoA", "Nitrospira", "Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus")

f <- amp_subset_taxa(amp, tax_vector = tax_vector_OTUs, remove = TRUE, normalise = TRUE)
#x<-f$tax

f$tax <- f$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level


### Coordinates ####
#long: 9.87335767946206
#Latitude : 56.8045096958576

#### Area to target ###
#56.800284, 9.877919
#56.808145, 9.877070
#56.808643, 9.861562
#56.802042, 9.859138

str_oks <- amp_subset_samples(f,
  longitude >= 9.859 & longitude <= 9.878,
                      latitude >= 56.800 & latitude <= 56.809, normalise=FALSE)

string<-str_oks$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)


heatmap_str_oks <- amp_heatmap(str_oks,
  facet_by = "mfd_hab1",
 group_by = "fieldsample_barcode",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("white", "#a71b1b"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 4,
 min_abundance = 0.01,
 round=1,
  order_y_by = rev(genus_string),
 normalise=FALSE) +
theme(axis.text.x = element_text(angle = 45, size=12, vjust = 1),
 axis.text.y = element_text(size=12),
  legend.position="right")+
  labs(title="Store Økssø")+
  theme(plot.title = element_text(hjust=0.5, face="bold", size = 14))


ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/store_oksso.png",
       heatmap_str_oks,
        height = 10,
  width = 11.5)

```

### Lille vildmose #####
```{r}



tax_vector_OTUs <- c("normalising_factor","Root; Homologues_mmoX","Root; likely_mmoX","Root_mmoX","Root_pmoA","amoA", "Nitrospira", "Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus")

f <- amp_subset_taxa(amp, tax_vector = tax_vector_OTUs, remove = TRUE, normalise = TRUE)
#x<-f$tax

f$tax <- f$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level



#### Area to target ###
#56.886808, 10.241401
#56.867573, 10.260944
#56.859278, 10.234604
#56.879911, 10.197461

lil_vild <- amp_subset_samples(f,
  longitude >= 10.197 & longitude <= 10.26,
    latitude >= 56.859 & latitude <= 56.8868, normalise=FALSE)

string<-lil_vild$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)


heatmap_lil_vild <- amp_heatmap(lil_vild,
  facet_by = "mfd_hab1",
 group_by = "fieldsample_barcode",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("white", "#a71b1b"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 4,
 min_abundance = 0.01,
 round=1,
 order_y_by = rev(genus_string),
 normalise=FALSE) +
theme(axis.text.x = element_text(angle = 45, size=12, vjust = 1),
 axis.text.y = element_text(size=12),
  legend.position="right")+
  labs(title="Lille Vildmose")+
  theme(plot.title = element_text(hjust=0.5, face="bold", size = 14))

ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/lille_vildmose.png",
       heatmap_lil_vild,
        height = 10,
  width = 7.5)

```


### Klosterheden Forest Plantation #####
```{r}



tax_vector_OTUs <- c("normalising_factor","Root; Homologues_mmoX","Root; likely_mmoX","Root_mmoX","Root_pmoA","Homologous_pmoA","Nitrosococcus", "Actinobacteria", "Probably_hydrocarbon_monooxygenases", "Cycloclasticus")

f <- amp_subset_taxa(amp, tax_vector = tax_vector_OTUs, remove = TRUE, normalise = TRUE)
#x<-f$tax

f$tax <- f$tax %>%
  mutate(Genus = str_extract(OTU, "[^;]*$")) ###Just for easy plotting at "genus = last word" level



#### Area to target ###
#56.49316, 8.43376
#56.49372, 8.339
#56.49372, 8.339
#56.44423, 8.43479

klosterheden <- amp_subset_samples(f,
  longitude >= 8.3 & longitude <= 8.47,
    latitude >= 56.42 & latitude <= 56.5, normalise=FALSE)

string<-klosterheden$tax %>%
  select(Genus | Species)
genus_string<-arrange(string, Species, decreasing=FALSE)
genus_string<-unlist(genus_string[,"Genus"])
genus_string <- unique(genus_string)


heatmap_klosterheden <- amp_heatmap(klosterheden,
  facet_by = "mfd_hab2",
 group_by = "fieldsample_barcode",
 tax_aggregate = "Genus",
 tax_show = 27,
 color_vector = c("white", "#a71b1b"),
 plot_colorscale = "sqrt",
 plot_values = TRUE,
 plot_values_size = 4,
 min_abundance = 0.1,
 round=2,
 order_y_by = rev(genus_string),
 normalise=FALSE) +
theme(axis.text.x = element_text(angle = 45, size=12, vjust = 1),
 axis.text.y = element_text(size=12),
  legend.position="right")+
  labs(title="Klosterheden Forest Plantation")+
  theme(plot.title = element_text(hjust=0.5, face="bold", size = 14))
heatmap_klosterheden
ggsave("/user_data/kalinka/GraftM/shallow_21_12_2022/R_scripts_and_figures/heatmap_klosterheden.png",
       heatmap_klosterheden,
        height = 10,
  width = 7.5)

```